python/ray/data/_internal/execution/operators/map_operator.py

```diff
@@ -322,6 +322,23 @@ def _get_runtime_ray_remote_args(
             new_remote_args = self._ray_remote_args_fn()
             for k, v in new_remote_args.items():
                 ray_remote_args[k] = v
+
+        if "memory" not in ray_remote_args:
+            # Typically, this configuration won't make a difference because
+            # `average_bytes_per_output` is usually ~128 MiB and each core usually has
+            # 4 GiB of memory. However, if `num_cpus` is small (e.g., 0.01) or
+            # `target_max_block_size` is large (e.g., 1GB), then tasks can OOM even
+            # if it just uses enough memory to produce an output block. By setting
+            # `memory` to the average output size, we can mitigate this case.
+            #
+            # We set it to 1 target block size out of assumption that *at least* 1 copy
+            # of data (to process heap) will be made during processing.
+            #
+            # Note that Ray's "memory" resource is exclusive of the Object Store
+            # memory allocated on the node (ie its total allocatable value is Total
+            # memory - Object Store memory).
+            ray_remote_args["memory"] = self.metrics.average_bytes_per_output
```

Comment:
this can be None. not sure if Ray Core can accept None. but better to avoid that.